nninit.lua

```diff
@@ -270,4 +270,95 @@ nninit.sparse = function(module, tensor, sparsity)
+  if filterRows ~= filterCols then
+    error("nninit.convaware requires the filters to have the same number of rows and columns, feel free to issue a pull request to extend this implementation")
+  end
+
+  -- Calculate "fanIn" and "fanOut" for 2d convolution tensors based on module conventions
+  local fanIn = filterStacks * filterRows * filterCols
+  local fanOut = filterCount * filterRows * filterCols
+
+  -- Setup options where "std" specifies the noise to break symmetry in the inverse Fourier transform
+  options = options or {}
+  std = options.std or 0.05
+
+  -- Specify the variables for the frequency domain tensor
+  local fourierTensor = require('signal').rfft2(torch.zeros(filterRows, filterCols))
+  local fourierRows = fourierTensor:size(1)
+  local fourierCols = fourierTensor:size(2)
+  local fourierSize = fourierRows * fourierCols
+
+  -- Specify the variables for the orthogonal tensor buffer
+  local orthogonalIndex = fourierSize
+  local orthogonalTensor = nil
+
+  -- For each filter, create a suitable basis tensor and perform an inverse Fourier transform to obtain the filter coefficients
+  for filterIndex = 1, filterCount do
+    basisTensor = torch.zeros(filterStacks, fourierSize)
+
+    -- Create a suitable basis tensor using the orthogonal tensor buffer, making sure to refill the buffer should it be empty
+    for basisIndex = 1, filterStacks do
+      if orthogonalIndex == fourierSize then
+        local randomTensor = torch.zeros(fourierSize, fourierSize):normal(0.0, 1.0)
+        local symmetricTensor = randomTensor + randomTensor:t() - torch.diag(randomTensor:diag())
+
+        orthogonalIndex = 0
+        orthogonalTensor, _, _ = torch.svd(symmetricTensor)
+      end
+
+      -- Copy a column from the orthogonal tensor buffer into the basis tensor
+      orthogonalIndex = orthogonalIndex + 1
+      basisTensor[{ { basisIndex }, {} }] = orthogonalTensor[{ {}, { orthogonalIndex } }]
+    end
+
+    basisTensor = basisTensor:view(filterStacks, fourierRows, fourierCols)
+
+    -- Perform the inverse Fourier transform from the basis tensor to obtain the filter coefficients, making sure to break the symmetry
+    for basisIndex = 1, filterStacks do
+      fourierTensor[{ {}, {}, { 1 } }] = basisTensor[{ { basisIndex }, {} }]
+      fourierTensor[{ {}, {}, { 2 } }]:zero()
+
+      -- Unlike the Numpy implementation, the inverse Fourier transform in the signal library does sadly only support a single size argument
+      tensor[{ { filterIndex }, { basisIndex }, {}, {} }] = require('signal').irfft2(fourierTensor, filterRows) + torch.zeros(filterRows, filterCols):normal(0.0, std)
+    end
+
+    -- Clear the orthogonal tensor buffer, we do not want to reuse it for the next filter
+    orthogonalIndex = fourierSize
+    orthogonalTensor = nil
+  end
+
+  -- Scale the filter variance to match the variance scheme defined by He-normal initialization
+  tensor:mul(torch.sqrt((2.0 / fanIn) * (1.0 / tensor:var())))
```

Comment:
Yes please - I've just added a note for Kaiming initialisation in the README which makes this clear, so you can simply copy this sentence over to make it clear what's going on. Although ReLUs are super popular, controlling `gain` manually mean that this initialisation is still theoretically sound when used with other types of nonlinearities. The issue may be a lack of knowledge of this, but the above line in the documentation makes it clear.